acme/challenge.go

```diff
@@ -360,25 +362,40 @@ type wireOidcPayload struct {
 func wireOIDC01Validate(ctx context.Context, ch *Challenge, db DB, jwk *jose.JSONWebKey, payload []byte) error {
 	prov, ok := ProvisionerFromContext(ctx)
 	if !ok {
-		return NewErrorISE("no provisioner provided")
+		return NewErrorISE("missing provisioner")
 	}
 
 	var oidcPayload wireOidcPayload
 	err := json.Unmarshal(payload, &oidcPayload)
 	if err != nil {
-		return storeError(ctx, db, ch, false, WrapError(ErrorRejectedIdentifierType, err,
-			"error unmarshalling Wire challenge payload"))
+		return WrapError(ErrorMalformedType, err, "error unmarshalling Wire OIDC challenge payload")
+	}
+
+	wireID, err := wire.ParseID([]byte(ch.Value))
+	if err != nil {
+		return WrapErrorISE(err, "error unmarshalling challenge data")
 	}
 
 	wireOptions, err := prov.GetOptions().GetWireOptions()
 	if err != nil {
 		return WrapErrorISE(err, "failed getting Wire options")
 	}
 
+	// TODO(hs): move this into validation below?
+	expectedKeyAuth, err := KeyAuthorization(ch.Token, jwk)
+	if err != nil {
+		return WrapErrorISE(err, "error determining key authorization")
+	}
+	if expectedKeyAuth != oidcPayload.KeyAuth {
+		return storeError(ctx, db, ch, true, NewError(ErrorRejectedIdentifierType,
+			"keyAuthorization does not match; expected %q, but got %q", expectedKeyAuth, oidcPayload.KeyAuth))
```

Comment:
In this case it's stored with the challenge in the DB, and will be returned to the client in case it's not nil. The behavior is the same as for HTTP validation. I don't think just the key authorization will be sufficient to do any sensitive without having access to the account private key, but I can change the error message slightly to not return the right value?